EventTiming: feature flag and stubs (https://w3c.github.io/event-timing)

francovs opened the pull request:
#### f33a25d11c72268ec627e75a1d1b699aaf297ae4
<pre>
EventTiming: feature flag and stubs (<a href="https://w3c.github.io/event-timing)">https://w3c.github.io/event-timing)</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=296142">https://bugs.webkit.org/show_bug.cgi?id=296142</a>
<a href="https://rdar.apple.com/156081762">rdar://156081762</a>

Reviewed by Ryan Reno.

Stubs out an implementation of event timing and adds a feature flag (EventTimingEnabled).

* LayoutTests/TestExpectations: skip tests that are timing out
* LayoutTests/platform/wpe/TestExpectations: skip tests that crash because of missing browser automation
* LayoutTests/platform/gtk/TestExpectations: skip tests that crash because of missing browser automation
* LayoutTests/platform/ios/TestExpectations: skip test that times out (missing browser automation?)
* LayoutTests/imported/w3c/web-platform-tests/event-timing/: reset test expectations
* Source/WebCore/page/EventCounts.[cpp|h|idl]: new files
* Source/WebCore/page/Performance+EventCounts.idl: new file
* Source/WebCore/page/PerformanceEventTiming.[cpp|h|idl]: new files
* Source/WebCore/page/Performance.[cpp|h]: moved code to fix build issues
* Source/WebCore/page/scrolling/ScrollLatchingController.h: fix build
* Source/WebCore/page/CaptionUserPreferencesMediaAF.cpp: fix build
* Source/WebCore/page/ios/ContentChangeObserver.cpp: fix build

Canonical link: <a href="https://commits.webkit.org/297683@main">https://commits.webkit.org/297683@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/156ff8d52e7e1c6f813ca3f83f80987cd35e3251

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/112513 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/32245 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/22723 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/118711 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/62964 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/114475 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/32897 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/40808 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/85630 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/36241 "") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/115460 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/26249 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/101219 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/65932 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/25547 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/19352 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/62471 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/105027 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/95638 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/19426 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/121933 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/111127 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/39587 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/29478 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/94496 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/39969 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/97452 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/94236 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24064 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/39347 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/17150 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/35661 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/39475 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/44963 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/135359 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/39112 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/36373 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/42447 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/40853 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -2674,6 +2674,21 @@ EventHandlerDrivenSmoothKeyboardScrollingEnabled:
 
+EventTimingEnabled:
+  type: bool
+  status: internal
```
testable is probably what you want, because you want start getting test rebaselines

francovs replied:
Enabling the flag currently is causing some tests in `web-platform-tests/event-timing/` to timeout instead of failing early, making the entire suite take over 20mins (from 25s). Because of that, ~I'm leaning towards keeping this `internal` (or maybe `unstable`) until that is sorted out~ will follow @rreno 's suggestion to skip and rebaseline the offending tests

nt1m replied:
You can just skip the tests for now in LayoutTests/TestExpectations

nt1m commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -2674,6 +2674,21 @@ EventHandlerDrivenSmoothKeyboardScrollingEnabled:
+  status: internal
+  category: dom
+  humanReadableName: "Event Timing API"
+  humanReadableDescription: "Enables the Event Timing API and supporting instrumentation"
```
```suggestion
  humanReadableDescription: "Enable the Event Timing API and supporting instrumentation"
```

nt1m commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -2674,6 +2674,21 @@ EventHandlerDrivenSmoothKeyboardScrollingEnabled:
+  category: dom
+  humanReadableName: "Event Timing API"
+  humanReadableDescription: "Enables the Event Timing API and supporting instrumentation"
+  webcoreGetter: isEventTimingEnabled
```
This should not be needed

nt1m reviewed (commented):
I don't think the copyright header is correct

rreno commented on Source/WebCore/page/EventCounts.cpp:
```diff
@@ -0,0 +1,39 @@
+
+void EventCounts::initializeMapLike(DOMMapAdapter& map)
+{
+    (void) map;
```
WebKit either uses ASSERT_UNUSED or in this case just omit the variable name in the argument list and the compiler won't complain about and unused variable.

rreno commented on Source/WebCore/page/EventCounts.h:
```diff
@@ -0,0 +1,49 @@
+public:
+
+    static Ref<EventCounts> create()
+    {
```
nit: this can be a one-liner.

rreno commented on Source/WebCore/page/PerformanceEventTiming.cpp:
```diff
@@ -0,0 +1,63 @@
+#include "config.h"
+#include "PerformanceEventTiming.h"
+
+#include "page/PerformanceEntry.h"
```
I think this include is redundant since it comes along for the ride in PerformanceEventTiming.h

rreno commented on Source/WebCore/page/PerformanceEventTiming.idl:
```diff
@@ -0,0 +1,40 @@
+    readonly attribute DOMHighResTimeStamp processingEnd;
+    readonly attribute boolean cancelable;
+    readonly attribute Node? target;
+    readonly attribute unsigned long long interactionId;  // Not implemented yet
```
I think none of this is implemented yet? Comment seems unnecessary.

nt1m commented on Source/WebCore/page/EventCounts.cpp:
```diff
@@ -0,0 +1,39 @@
+#include "config.h"
+#include "EventCounts.h"
+
+#include <bindings/js/JSDOMMapLike.h>
```
```suggestion
#include "JSDOMMapLike.h"
```

francovs committed:
EventTiming: feature flag and stubs (https://w3c.github.io/event-timing)
https://bugs.webkit.org/show_bug.cgi?id=296142
rdar://156081762

Reviewed by Ryan Reno.

Stubs out an implementation of event timing and adds a feature flag (EventTimingEnabled).

* LayoutTests/TestExpectations: skip tests that are timing out
* LayoutTests/platform/wpe/TestExpectations: skip tests that crash because of missing browser automation
* LayoutTests/platform/gtk/TestExpectations: skip tests that crash because of missing browser automation
* LayoutTests/platform/ios/TestExpectations: skip test that times out (missing browser automation?)
* LayoutTests/imported/w3c/web-platform-tests/event-timing/: reset test expectations
* Source/WebCore/page/EventCounts.[cpp|h|idl]: new files
* Source/WebCore/page/Performance+EventCounts.idl: new file
* Source/WebCore/page/PerformanceEventTiming.[cpp|h|idl]: new files
* Source/WebCore/page/Performance.[cpp|h]: moved code to fix build issues
* Source/WebCore/page/scrolling/ScrollLatchingController.h: fix build
* Source/WebCore/page/CaptionUserPreferencesMediaAF.cpp: fix build
* Source/WebCore/page/ios/ContentChangeObserver.cpp: fix build

Canonical link: https://commits.webkit.org/297683@main
